addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+        ],
+        string="Refund State",
+        default="draft",
+        readonly=True,
+        copy=False,
+    )
+    l10n_tw_edi_refund_agreement_type = fields.Selection(
+        [("offline", "Offline Agreement"), ("online", "Online Agreement")],
+        default="offline",
+        string="Refund invoice Agreement Type",
+        required=True,
+        copy=False,
+    )
+    l10n_tw_edi_origin_invoice_number = fields.Many2one('account.move', string="Original Invoice", readonly=True, copy=False)
+    l10n_tw_edi_refund_invoice_number = fields.Char(string="Refund Invoice Number", readonly=True, copy=False)
+    l10n_tw_edi_remain_refundable_amount = fields.Float(string="Remain refundable amount", readonly=True, copy=False)
+
+    @api.depends('l10n_tw_edi_is_donate', 'l10n_tw_edi_carrier_type')
+    def _compute_is_print(self):
+        for move in self:
+            if move.l10n_tw_edi_is_donate or move.l10n_tw_edi_carrier_type:
+                move.l10n_tw_edi_is_print = False
+
+    @api.depends('l10n_tw_edi_is_print', 'l10n_tw_edi_carrier_type')
+    def _compute_is_donate(self):
+        for move in self:
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_carrier_type:
+                move.l10n_tw_edi_is_donate = False
+                move.l10n_tw_edi_love_code = False
+
+    @api.depends('l10n_tw_edi_is_print', 'l10n_tw_edi_is_donate')
+    def _compute_carrier_info(self):
+        for move in self:
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_is_donate:
+                move.l10n_tw_edi_carrier_type = False
+                move.l10n_tw_edi_carrier_number = False
+
+    @api.depends('line_ids.tax_ids')
+    def _compute_is_included_tax(self):
+        for move in self:
+            product_lines = move.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines())
+            move.l10n_tw_edi_ecpay_is_included_tax = "1" if any(tax.amount >= 5.0 for line in product_lines for tax in line.tax_ids) else "0"
+
+    # ----------------
+    # Business methods
+    # ----------------
+
+    # API methods
+
+    def _l10n_tw_edi_prepare_item_list(self, json_data):
+        item_list = []
+        for line in self.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines()):
+            taxable = line.tax_ids.filtered(lambda t: t.amount >= 5.0)
+            item_list.append(
+                {
+                    "ItemSeq": line.sequence,
+                    "ItemName": line.product_id.name[:100],
+                    "ItemCount": int(line.quantity),
+                    "ItemWord": line.product_uom_id.name[:6],
+                    "ItemPrice": int(line.price_total / line.quantity),
+                    "ItemTaxType": "1" if len(taxable.ids) >= 1 else "3",
+                    "ItemAmount": int(line.price_total),
+                }
+            )
+        json_data.update({"Items": item_list})
```

Comment:
```suggestion
        json_data['Items'] = item_list
```
It's as readable, if not more, and more performant